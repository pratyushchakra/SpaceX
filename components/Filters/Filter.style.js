import css from 'styled-jsx/css';

const FilterStyles = css`
.filters {
    padding: 10px;
    font-size: 20px;
    font-weight: bold;
}
.filter-name-text {
    text-align: center;
}
.date-btns {
    display: grid;
    grid-template-columns: 50% 50%;
}
.border-bottom {
    border-bottom: 2px solid #e2dfe0;
    margin: 2px 40px 4px 40px;
}
button {
    color: #000;
    background: #c5e09b;
    border-radius: 2px;
    width: 65px;
    height: 25px;
}
.button-wrapper {
    margin: 0 auto;
    margin-top: 7px;
    margin-bottom: 7px;
}
.selected {
    background: #7cba01;
}
`
export default FilterStyles;