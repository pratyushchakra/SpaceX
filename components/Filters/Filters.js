import { Fragment } from "react";

import { withContext } from '../../utils/Context';
import { generateYears } from "../../utils/utils";
import style from './Filter.style'
const Filters = ({ store }) => {
    const { filterConfig: { filterData, setFilterData } } = store;
    const isLaunchAndLandSelected = () => {
        const { launch, landing } = filterData;
        const isLaunchSelected = typeof launch === 'boolean';
        const isLandingSelected = typeof landing === 'boolean';
        return isLaunchSelected && isLandingSelected;
    }
    const changeFilterConfig = (label, value) => () => {
        let tempValue = value;
        const { launch, landing } = filterData;
        const isLaunchSelected = typeof launch === 'boolean';
        const isLandingSelected = typeof landing === 'boolean';
        const config = {
            ...filterData
        }
        if (filterData[label] === value) {
            tempValue = null;
        }
        if (label === 'year' && value && !isLaunchSelected && !isLandingSelected) {
            config.launch = true;
            config.landing = true;
        }
        config[label] = tempValue;
        setFilterData(config)
    }
    return (
        <Fragment>
            <p className="filters">Filters</p>
            <div className="date-filter">
                <p className="filter-name-text">Launch Year</p>
                <div className="border-bottom" />
                <div className="date-btns">
                    {
                        generateYears().map(y => (
                            <div
                                key={y}
                                className="button-wrapper">
                                <button
                                    className={filterData['year'] === y ? 'selected' : ''}
                                    onClick={changeFilterConfig('year', y)}
                                >{y}</button>
                            </div>
                        ))
                    }
                </div>
            </div>
            <div className="launch-filter">
                <p className="filter-name-text">Successful Launch</p>
                <div className="border-bottom" />
                <div className="date-btns">
                    <div className="button-wrapper">
                        <button
                            className={filterData['launch'] === true ? 'selected' : ''}
                            onClick={changeFilterConfig('launch', true)}>True</button>
                    </div>
                    <div className="button-wrapper">
                        <button
                            className={filterData['launch'] === false ? 'selected' : ''}
                            onClick={changeFilterConfig('launch', false)}>False</button>
                    </div>
                </div>
            </div>
            <div className="landing-filter">
                <p className="filter-name-text">Successfull Landing</p>
                <div className="border-bottom" />
                <div className="date-btns">
                    <div className="button-wrapper">
                        <button
                            className={filterData['landing'] === true ? 'selected' : ''}
                            onClick={changeFilterConfig('landing', true)}>True</button>
                    </div>
                    <div className="button-wrapper">
                        <button
                            className={filterData['landing'] === false ? 'selected' : ''}
                            onClick={changeFilterConfig('landing', false)}>False</button>
                    </div>
                </div>
            </div>
            <style jsx>{style}</style>
        </Fragment>
    )
};

export default withContext(Filters)