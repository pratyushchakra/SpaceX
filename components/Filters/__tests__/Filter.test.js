import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { act } from 'react-dom/test-utils'

import { Filters } from '../Filters';
import mock from '../../../utils/mock';
Enzyme.configure({ adapter: new Adapter() });

describe('Filter Component', () => {
    let store
    beforeEach(() => {
        store = {
            "launchConfig": {
                "allLaunches": mock,
                "setLaunchData": jest.fn()
            },
            "filterConfig": {
                "filterData": {},
                "setFilterData": jest.fn()
            }
        }
    })
    it('renders properly with valid data', () => {
        const wrapper = mount(
            <Filters store={store} />
        );
        expect(wrapper.debug()).toMatchSnapshot();
    });
    it('selects filter and highligts them', async () => {
        const wrapper = mount(
            <Filters store={store} />
        );
        await act(async () => {
            wrapper.find('#year-btn-2010').simulate('click');
            await jest.runAllTimers()
            await wrapper.update()
        })
        await act(async () => {
            await jest.runAllTimers()
            await wrapper.update()
        });
        expect(wrapper.debug()).toMatchSnapshot();
        // console.log('wrapper.find', wrapper.find('#year-btn-2010').html());
        // console.log('wrapper.selvet', wrapper.find('selected').length);
        // expect(wrapper.find('#year-btn-2010').hasClass('selected')).to.equal(true);
    })
})