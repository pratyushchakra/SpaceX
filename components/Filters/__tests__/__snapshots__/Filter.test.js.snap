// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component renders properly with valid data 1`] = `
"<Filters store={{...}}>
  <p className=\\"jsx-1376872141 filters\\">
    Filters
  </p>
  <div className=\\"jsx-1376872141 date-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Launch Year
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2008\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2008
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2009\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2009
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2010\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2010
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2011\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2011
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2012\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2012
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2013\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2013
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2014\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2014
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2015\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2015
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2016\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2016
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2017\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2017
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2018\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2018
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2019\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2019
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2020\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2020
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2021\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2021
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2022\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2022
        </button>
      </div>
    </div>
  </div>
  <div className=\\"jsx-1376872141 launch-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Successful Launch
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          True
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          False
        </button>
      </div>
    </div>
  </div>
  <div className=\\"jsx-1376872141 landing-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Successfull Landing
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          True
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          False
        </button>
      </div>
    </div>
  </div>
  <JSXStyle id=\\"1376872141\\" />
</Filters>"
`;

exports[`Filter Component selects filter and highligts them 1`] = `
"<Filters store={{...}}>
  <p className=\\"jsx-1376872141 filters\\">
    Filters
  </p>
  <div className=\\"jsx-1376872141 date-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Launch Year
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2008\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2008
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2009\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2009
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2010\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2010
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2011\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2011
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2012\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2012
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2013\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2013
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2014\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2014
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2015\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2015
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2016\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2016
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2017\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2017
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2018\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2018
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2019\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2019
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2020\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2020
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2021\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2021
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button id=\\"year-btn-2022\\" onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          2022
        </button>
      </div>
    </div>
  </div>
  <div className=\\"jsx-1376872141 launch-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Successful Launch
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          True
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          False
        </button>
      </div>
    </div>
  </div>
  <div className=\\"jsx-1376872141 landing-filter\\">
    <p className=\\"jsx-1376872141 filter-name-text\\">
      Successfull Landing
    </p>
    <div className=\\"jsx-1376872141 border-bottom\\" />
    <div className=\\"jsx-1376872141 date-btns\\">
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          True
        </button>
      </div>
      <div className=\\"jsx-1376872141 button-wrapper\\">
        <button onClick={[Function (anonymous)]} className=\\"jsx-1376872141 \\">
          False
        </button>
      </div>
    </div>
  </div>
  <JSXStyle id=\\"1376872141\\" />
</Filters>"
`;
